src/rez/solver.py

```diff
@@ -1499,6 +1881,9 @@ def _get_solved_variants(self):
         for scope in self.scopes:
             variant = scope._get_solved_variant()
             if variant:
+                # We changed the index to help the solver pick the 'preferred' variant
+                # At this point we should add it back for the original index
+                # variant.index = variant.userdata.variables.get('index', None)
```

Comment:
This line was needed to restore the original index, but by doing  here in the new `resource2` branch it ends up with some inconsistencies on the .rxt file. So  I have now also removed from here and at the end of  `Solver.solve_step()` (line 2139) I made a call to a new function called  `self._revert_sorted_indexes()` which reads

```
+    def _revert_sorted_indexes(self):
+        # We changed the index to help the solver pick the 'preferred' variant
+        # Restoring the original index from the userdata (resource handle)
+
+        # Get the last (succeeded or failed) phase
+        final_phase = self.phase_stack[-1]
+
+        for scope in final_phase.scopes:
+            variant = scope._get_solved_variant()
+            if variant:
+                variant.index = variant.userdata.get('variables').get('index', None)
```